Sources/Networking/HTTPClient/DNSChecker.swift

```diff
@@ -45,18 +45,18 @@ enum DNSChecker: DNSCheckerType {
     }
 
     static func errorWithBlockedHostFromError(_ error: Error?) -> NetworkError? {
-        guard isBlockedAPIError(error),
+        guard self.isBlockedAPIError(error),
               let nsError = error as NSError?,
               let failedURL = nsError.userInfo[NSURLErrorFailingURLErrorKey] as? URL else {
             return nil
         }
 
-        let host = resolvedHost(fromURL: failedURL)
+        let host = self.resolvedHost(fromURL: failedURL)
         return .dnsError(failedURL: failedURL, resolvedHost: host)
     }
 
     static func isBlockedURL(_ url: URL) -> Bool {
```

Comment:
Should we make this private now? I believe it's not called outside this class other than it tests and we can test the public API for that.

Reply:
Oh yeah let me check.

Reply:
We have specific tests for this method, which I think is useful.

Reply:
Well, I try to avoid making things public only for tests. I think if it needs to be tested because the logic is too complex and the public API of the class does too many things, it should be extracted to a different class where it can be tested better. This doesn't seem to be that complex of a logic though, so I don't think it warrants moving to a different class.

That said, I don't think this should block the PR.

Reply:
> Well, I try to avoid making things public only for tests. I think if it needs to be tested because the logic is too complex and the public API of the class does too many things, it should be extracted to a different class where it can be tested better.

💯

Reply:
I agree. The purpose of this PR was to avoid calling the DNS logic twice though.